src/html-conventions-lib/class-attributes-with-same-name.xsl

```diff
@@ -0,0 +1,159 @@
+                    <xsl:copy-of select="$attributeChecks"/>
+                </xsl:if>
+            </xsl:if>
+        </xsl:for-each>
+    </xsl:template>
+
+
+
+    <xd:doc>
+        <xd:desc>Check the multiplicity values from a group of class attributes with same
+            name</xd:desc>
+        <xd:param name="attributeName"/>
+        <xd:param name="root"/>
+    </xd:doc>
+    <xsl:template name="checkMultiplicityOfAttributesWithSameName">
+        <xsl:param name="attributeName"/>
+        <xsl:param name="root"/>
+        <xsl:variable name="attributesWithSameName"
+            select="f:getClassAttributeByName($attributeName, $root)"/>
+        <xsl:variable name="lowerBoundValues" select="$attributesWithSameName/bounds/@lower"/>
+        <xsl:variable name="upperBoundValues" select="$attributesWithSameName/bounds/@upper"/>
+        <xsl:variable name="allAttributesHaveMultiplicityValue"
+            select="fn:count($attributesWithSameName) = fn:count($lowerBoundValues) and fn:count($lowerBoundValues) = fn:count($upperBoundValues)"/>
+        <xsl:sequence
+            select="
+                if (f:areStringsEqual($lowerBoundValues) and f:areStringsEqual($upperBoundValues) and $allAttributesHaveMultiplicityValue) then
+                    ()
+                else
+                    f:generateHtmlWarning('When a property is reused in multiple contexts, the multiplicity is expected to be the same. Please check the nomenclature above for a summary.')"
+        />
+    </xsl:template>
+
+
+    <xd:doc>
+        <xd:desc>Check the definition values from a group of class attributes with same
+            name</xd:desc>
+        <xd:param name="attributeName"/>
+        <xd:param name="root"/>
+    </xd:doc>
+    <xsl:template name="checkDefinitionOfAttributesWithSameName">
+        <xsl:param name="attributeName"/>
+        <xsl:param name="root"/>
+        <xsl:variable name="attributesWithSameName"
+            select="f:getClassAttributeByName($attributeName, $root)"/>
+        <xsl:variable name="definitionValues" select="$attributesWithSameName/documentation/@value"/>
+        <xsl:variable name="descriptionsWithAnnotations" as="xs:string*"
+            select="
+                for $attribute in $attributesWithSameName
+                return
+                    if ($attribute/documentation/@value) then
+                        fn:concat($attribute/documentation/@value, ' (', $attribute/../../@name, ') ')
+                    else
+                        ()"/>
+
+        <xsl:variable name="allAttributesHaveDefinition"
+            select="fn:count($attributesWithSameName) = fn:count($definitionValues)"/>
+        <xsl:sequence
+            select="
+                if (f:areStringsEqual($definitionValues) and fn:boolean($definitionValues) and $allAttributesHaveDefinition) then
+                    f:generateHtmlInfo(fn:concat('The property is reused in multiple contexts, the meaning given by the definition is the same.  ',
```

Comment:
do we need this info? I would say we don't. This is the norm.

Reply:
nope